easybuild/easyblocks/x/xmipp.py

```diff
@@ -25,174 +25,203 @@
 """
 EasyBuild support for building and installing Xmipp, implemented as an easyblock
 
-@author: Jens Timmerman (Ghent University)
-@author: Pablo Escobar (sciCORE, SIB, University of Basel)
-@author: Kenneth Hoste (Ghent University)
+@author: Ake Sandgren (HPC2N, Umea University)
 """
-import fileinput
 import os
-import re
-import stat
-import sys
 
 import easybuild.tools.environment as env
-import easybuild.tools.toolchain as toolchain
-from easybuild.easyblocks.generic.pythonpackage import det_pylibdir
-from easybuild.framework.easyblock import EasyBlock
-from easybuild.framework.easyconfig import CUSTOM
+from distutils.version import LooseVersion
+from easybuild.easyblocks.generic.scons import SCons
 from easybuild.tools.build_log import EasyBuildError
-from easybuild.tools.filetools import adjust_permissions, mkdir, write_file
+from easybuild.tools.filetools import apply_regex_substitutions, change_dir
+from easybuild.tools.filetools import copy_file, mkdir, remove_file, symlink
 from easybuild.tools.modules import get_software_root, get_software_version
 from easybuild.tools.run import run_cmd
+from easybuild.tools.systemtools import get_shared_lib_ext
 
 
-class EB_Xmipp(EasyBlock):
-    """
-    easyblock to install Xmipp
-    """
+class EB_Xmipp(SCons):
+    """Support for building/installing Xmipp."""
 
     def __init__(self, *args, **kwargs):
-        """Easyblock constructor, enable building in installation directory."""
+        """Initialize Xmipp-specific variables."""
         super(EB_Xmipp, self).__init__(*args, **kwargs)
-        self.build_in_installdir = True
-        self.xmipp_pythonpaths = []
+        self.xmipp_modules = ['xmippCore', 'xmipp', 'xmippViz']
+        self.srcdir = os.path.join(self.builddir, 'src')
+        self.cfg['start_dir'] = self.builddir
+        self.use_cuda = False
 
     def extract_step(self):
         """Extract Xmipp sources."""
-        # strip off 'xmipp' part to avoid having everything in a 'xmipp' subdirectory
-        self.cfg.update('unpack_options', '--strip-components=1')
+        # Xmipp assumes that everything is unpacked in a "src" dir
+        mkdir(self.srcdir)
+        self.cfg.update('unpack_options', '--directory src')
```

Comment:
use `os.path.basename(self.srcdir)` rather than hardcoding `src`

```python
self.cfg.update('unpack_options', '--directory %s' % os.path.basename(self.srcdir))
```

Reply:
What do you mean? Xmipp assumes that it lives in a directory called "src", it's used in the scripts.

self.srcdir is not named src is it?

Reply:
Ahh, I had actually set it...